### ISSUE TITLE
before_post_process not run with rails 5

### ISSUE THREAD

**Issue by maletor** [thoughtbot/paperclip#2397]
I upgraded a project to rails 5 that is using paperclip 4.3.6.

I'm trying to debug why `before_post_process` is no longer called.

The issue comes up because it's trying to run `styles: { thumb: '104x84#' }` on non-images.

The code is

```ruby
before_post_process :image?

def image?
  !(attachment_content_type =~ /^image.*/).nil?
end
```

If we do `model.attachment = URI object`, `before_post_process` is never run. Is this a bug? Definitely seems like it.

**Comment by ztsuperman** [comment]
Same issue.

**Comment by benlangfeld** [comment]
This is fixed in Paperclip 5.1.0 at least (maybe earlier).
